Gems/AzQtComponentsForPython/Code/CMakeLists.txt

```diff
@@ -0,0 +1,146 @@
+    if(NOT python_interpreter)
+        find_program(python_interpreter "python")
+    endif()
+
+    if(NOT python_interpreter)
+        message(FATAL_ERROR "Error: Failed to locate Python installation.")
+    endif()
+    
+    # Load the library/exe locations. These are platform specific.
+    include(${pal_dir}/azqtcomponentsforpython_file_settings.cmake)
+    
+    #Define the shared library for which we will create bindings.
+    set_property(TARGET ${AZQTCOMPONENTS_LIBRARY} PROPERTY PREFIX "")
+
+    # Ensure the generated wrapper files have the correct header.
+    set(LICENSE_FILE ${CMAKE_CURRENT_LIST_DIR}/license_header.txt)
+
+    # Set up the options to pass to shiboken.
+    set(shiboken_options --generator-set=shiboken --enable-parent-ctor-heuristic
+        --enable-pyside-extensions --enable-return-value-heuristic --use-isnull-as-nb_nonzero
+        --avoid-protected-hack --language-level=c++17 --debug-level=full
+        --license-file=${LICENSE_FILE}
+        -I${LY_ROOT_FOLDER}/Code/Framework/AzCore
+        -I${LY_ROOT_FOLDER}/Code/Framework/AzQtComponents
+        -I${LY_ROOT_FOLDER}/Code/Framework/AzCore/Platform/${PAL_PLATFORM_NAME} 
+        -I${QT_MODULE_DIR}/include
+        -I${QT_MODULE_DIR}/include/QtWidgets
+        -I${QT_MODULE_DIR}/include/QtCore
+        -I${QT_MODULE_DIR}/include/QtGui
+        -T${PYSIDE2_TYPESYSTEM_PATH}
+        -T${PYSIDE2_SHARE_PATH}
+        --output-directory=${CMAKE_CURRENT_BINARY_DIR}
+    )
+
+    set(generated_sources_dependencies ${wrapped_header} ${typesystem_file})
+
+    add_custom_command(
+        OUTPUT ${WRAPPED_FILES}
+        COMMAND ${SHIBOKEN_GENERATOR_EXE_PATH} ${shiboken_options} ${wrapped_header} ${typesystem_file}
+        DEPENDS ${generated_sources_dependencies} 
+        COMMENT "Running generator for ${typesystem_file}."
+        VERBATIM
+    )
+
+    # Set up the project for building the code generated by shiboken.
+    ly_add_target(
+        NAME ${GEM_NAME} GEM_MODULE
+        NAMESPACE Gem
+        OUTPUT_NAME ${BINDINGS_LIBRARY}
+        FILES_CMAKE
+            wrapped_files.cmake
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                ${PYSIDE2_INCLUDE_DIR}
+                ${PYTHON_INCLUDE_DIR}
+                ${SHIBOKEN_INCLUDE_DIR}
+                ${PYSIDE2_INCLUDE_DIR}\\QtCore
+                ${PYSIDE2_INCLUDE_DIR}\\QtGui
+                ${PYSIDE2_INCLUDE_DIR}\\QtWidgets
+        BUILD_DEPENDENCIES
+            PUBLIC
+                3rdParty::pyside2
+                Qt5::Widgets
+                Qt5::Core
+                Qt5::Widgets
+                Qt5::Gui
+                ${AZQTCOMPONENTS_LIBRARY}
+                Gem::EditorPythonBindings.Static
+    )
+
+    ly_create_alias(NAME ${MODULE_NAME}.Builders NAMESPACE Gem TARGETS ${MODULE_NAME})
+    ly_create_alias(NAME ${MODULE_NAME}.Tools NAMESPACE Gem TARGETS ${MODULE_NAME})
+    ly_create_alias(NAME ${MODULE_NAME}.Clients NAMESPACE Gem TARGETS ${MODULE_NAME})
```

Comment:
We don't have QT available at runtime in the Luancher. therefore the .Clients target should be removed.

Furthermore the cmake logic needs to be put under an if([PAL_TRAITS_HOST_TOOLS](https://github.com/o3de/o3de/blob/development/cmake/Platform/Windows/PAL_windows.cmake#L13)) check to prevent this code from building on Android and IOS.

Reply:
Removed